### DISCUSSION TITLE
Arbitrary messages

### DISCUSSION THREAD

**Discussion by Tarang** [supabase/supabase#737] [Questions]
With the realtime project is it possible to send arbitrary messages through, for example to send a `hello` to everyone subscribed to a channel without it having to go through as a database update?

**Comment by kiwicopple** [comment]
It isn't, but this is something we will need to build. Can I ask what your use-case is?

**Reply by Tarang** [reply]
Hi,

Examples:
- One way (server originated) - Notifications (e.g log user out), 
- One way (client originated) - Event notification, e.g user clicked x
- Two way (client originated) - Internal method to forward to a http endpoint get the result and pass it back
- Two way (server originated) - Internal method to check if client is active and pass the result back

Also the one way methods are extremely high throughput so it would be wasteful for them to get an OK type response back.

**Reply by Tarang** [reply]
@kiwicopple if you help me understand the wire protocol i could implement them. There are 2 classes one is pub/sub (one way messages) and the other is method types

I worked on another protocol DDP which had the method types but no notification type, but you could make 'virtual tables' (not saved to disk only exists on the client) to create an implementation. Is something like this possible with Supabase now?

I'm keen to use whatever you have/a workaround since there's no proper implementation of these features yet. 

You might ask, why use Supabase and not some other socket for this. Using supabase would really make it cleaner and simpler. It's also elixir based so i have the confidence it can work well.
